app/src/main/java/com/owncloud/android/operations/SynchronizeFolderOperation.java

```diff
@@ -441,8 +446,30 @@ private void syncContents() throws OperationCancelledException {
     }
 
     private void startDirectDownloads() {
-        final var fileDownloadHelper = FileDownloadHelper.Companion.instance();
-        mFilesForDirectDownload.forEach(file -> fileDownloadHelper.downloadFile(user, file));
+        if (syncForInternalTwoWaySyncWorker) {
+            try {
+                for (OCFile file: mFilesForDirectDownload) {
+                    if (file == null) continue;
+
+                    final var operation = new DownloadFileOperation(user, file, mContext);
+                    var result = operation.execute(getClient());
+
+                    String filename = file.getFileName();
+                    if (filename == null) continue;
```

Comment:
```suggestion
if (filename == null)  {
   continue;
   }
```

better readable, and if we add any line it will not break logic.

Reply:
https://github.com/nextcloud/android/pull/13993/commits/c45d8394e25d6b960d55c92702132c6858242085